bluebutton-server-app/src/test/java/gov/hhs/cms/bluebutton/server/app/utils/EndpointJsonResponseComparatorIT.java

```diff
@@ -0,0 +1,762 @@
+				{ "patientSearchById", (Supplier<String>) EndpointJsonResponseComparatorIT::patientSearchById },
+				{ "patientByIdentifier", (Supplier<String>) EndpointJsonResponseComparatorIT::patientByIdentifier },
+				{ "coverageRead", (Supplier<String>) EndpointJsonResponseComparatorIT::coverageRead },
+				{ "coverageSearchByPatientId",
+						(Supplier<String>) EndpointJsonResponseComparatorIT::coverageSearchByPatientId },
+				{ "eobByPatientIdAll", (Supplier<String>) EndpointJsonResponseComparatorIT::eobByPatientIdAll },
+				{ "eobByPatientIdPaged", (Supplier<String>) EndpointJsonResponseComparatorIT::eobByPatientIdPaged },
+				{ "eobReadCarrier", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadCarrier },
+				{ "eobReadDme", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadDme },
+				{ "eobReadHha", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadHha },
+				{ "eobReadHospice", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadHospice },
+				{ "eobReadInpatient", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadInpatient },
+				{ "eobReadOutpatient", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadOutpatient },
+				{ "eobReadPde", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadPde },
+				{ "eobReadSnf", (Supplier<String>) EndpointJsonResponseComparatorIT::eobReadSnf } };
+	}
+
+	private final String endpointId;
+	private final Supplier<String> endpointOperation;
+
+	/**
+	 * Parameterized test constructor: JUnit will construct a new instance of this
+	 * class for every top-level element returned by the {@link #data()}
+	 * {@link Parameters} test data generator, and then run the test cases in this
+	 * class using that specific test data element.
+	 * 
+	 * @param endpointId
+	 *            the name of the operation being tested, which is also used to
+	 *            locate the "approved" operation response file in the
+	 *            src/test/resources/endpoint-responses source directory
+	 * @param endpointOperation
+	 *            the operation to be tested
+	 * 
+	 */
+	public EndpointJsonResponseComparatorIT(String endpointId, Supplier<String> endpointOperation) {
+		this.endpointId = endpointId;
+		this.endpointOperation = endpointOperation;
+	}
+
+	/**
+	 * Generates current endpoint response files, comparing them to the
+	 * corresponding approved responses.
+	 */
+	@Test
+	public void verifyCorrectEndpointResponse() {
+		Path targetResponseDir = getTargetResponseDir();
+
+		// Call the server endpoint and save its result out to a file corresponding to
+		// the endpoint Id.
+		String endpointResponse = endpointOperation.get();
+		writeFile(endpointResponse, generateFileName(targetResponseDir, endpointId));
+
+		assertJsonDiffIsEmpty(endpointId);
+	}
+
+	/**
+	 * Generates the "golden" files, i.e. the approved responses to compare to. Run
+	 * by commenting out the @Ignore annotation and running this method as JUnit.
```

Comment:
```suggestion
	 * by commenting out the <code>@Ignore</code> annotation and running this method as JUnit.
```

Not sure that's quite right, either. Does JavaDoc require you to escape @s?